File: hw9/java/MySQLTest2.java

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.PreparedStatement;
import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;
import java.util.Scanner;
import com.mysql.jdbc.Driver;

public class MySQLTest2 {

    public static void main(String[] args) {
	try {
	    // connection info
	    Properties prop = new Properties();
	    FileInputStream in = new FileInputStream("config.properties");
	    prop.load(in);
	    in.close();	    

	    // connect to datbase
	    String hst = prop.getProperty("host");
	    String usr = prop.getProperty("user");
	    String pwd = prop.getProperty("password");
	    String dab = "cpsc321";
	    String url = "jdbc:mysql://" + hst + "/" + dab;
	    Connection con = DriverManager.getConnection(url, usr, pwd);

	    // get a category from the user
	    System.out.print("Please enter a category id: ");
	    Scanner reader = new Scanner(System.in);
	    String userInput1 = reader.next();
	    System.out.print("Please enter a title keyword: ");
	    String userInput2 = reader.next();
	    reader.close();

	    // create and execute a prepared statement 
	    String q =
		"SELECT COUNT(*) as total FROM film_category " +
		"JOIN film USING (film_id) " + 
		"WHERE category_id = ? AND INSTR(title, ?)";
	    PreparedStatement stmt = con.prepareStatement(q);
	    stmt.setString(1, userInput1);
	    stmt.setString(2, userInput2.toUpperCase());
	    ResultSet rs = stmt.executeQuery();
	    
	    // print the result
	    if(rs.next()) {
		int num_films = rs.getInt("total");
		String ans = "There are " + num_films + " films with " +
		    "category id " + userInput1 + " and titles " + 
		    "containing " + userInput2.toUpperCase() + ".";
		System.out.println(ans);
	    }

	    // release resources
	    rs.close();
	    stmt.close();
	    con.close();
	    
	} catch(Exception err) {
	    err.printStackTrace();
	}
    }
    
}

File: hw9/python/mysql_test_2.py

import mysql.connector
import config

def main():
    try: 
        # connection info
        usr = config.mysql['user']
        pwd = config.mysql['password']
        hst = config.mysql['host']
        dab = 'cpsc321'
        # create a connection
        con = mysql.connector.connect(user=usr,password=pwd, host=hst,
                                      database=dab)
        # get a category from the user
        user_input_1 = raw_input('Please enter a category id: ')
        user_input_2 = raw_input('Please enter a title keyword: ') 

        # create and execute query
        rs = con.cursor()
        query = '''SELECT COUNT(*) as total 
                   FROM   film_category JOIN film USING (film_id) 
                   WHERE  category_id = %s AND INSTR(title, %s)'''
        rs.execute(query, (user_input_1, user_input_2.upper()))

        # print the result
        row = rs.fetchone()
        if row is not None:
            result = 'There are {} films with category id {} '
            result += 'and titles containing "{}".'
            result = result.format(row[0], user_input_1, user_input_2)
            print result

        rs.close()
        con.close()
            
    except mysql.connector.Error as err:
        print err


if __name__ == '__main__':
    main()
    
